File: POM/PolicyCenter/Policy_Cancel.js
import { expect } from "@playwright/test";
exports.PolicyCancel =
class PolicyCancel {

    constructor(page){
	this.page = page;
    this.actions='#PolicyFile\\:PolicyFileMenuActions-btnInnerEl';
	this.cancel_Policy='#PolicyFile\\:PolicyFileMenuActions\\:PolicyFileMenuActions_NewWorkOrder\\:PolicyFileMenuActions_CancelPolicy-textEl';
	this.source='#StartCancellation\\:StartCancellationScreen\\:CancelPolicyDV\\:Source-inputEl';
	this.reason='[id="StartCancellation:StartCancellationScreen:CancelPolicyDV:Reason-inputEl"]';
	this.reason_Description='#StartCancellation\\:StartCancellationScreen\\:CancelPolicyDV\\:ReasonDescription-inputEl';
	this.refund_Method='#StartCancellation\\:StartCancellationScreen\\:CancelPolicyDV\\:CalcMethod-inputEl';
	this.start_CancellationBtn='#StartCancellation\\:StartCancellationScreen\\:NewCancellation-btnInnerEl';
	this.bind_Option='#CancellationWizard\\:CancellationWizard_QuoteScreen\\:JobWizardToolbarButtonSet\\:BindOptions-btnInnerEl';
	this.cancel_Now='#CancellationWizard\\:CancellationWizard_QuoteScreen\\:JobWizardToolbarButtonSet\\:BindOptions\\:CancelNow-textEl';
	this.warning_Message_Ok="//*[text()='OK']";
	this.tittle_Bar_Status='#JobComplete\\:JobCompleteScreen\\:ttlBar';
	this.view_Your_Policy='#JobComplete\\:JobCompleteScreen\\:JobCompleteDV\\:ViewPolicy-inputEl';
	this.policy_Number='#PolicyFile_Summary\\:Policy_SummaryScreen\\:Policy_Summary_PolicyDV\\:PolicyNumber-inputEl';
	this.policy_Transactions='#PolicyFile\\:MenuLinks\\:PolicyFile_PolicyFile_Jobs';
	this.policy_Current_Status="//div[@id='PolicyFile_Jobs/DetailsTitleCard']//table//table[1]//tr//td[5]//div";

    }

	async policyCalcel(){

		await this.page.locator(this.actions).click();
		await this.page.locator(this.cancel_Policy).click();
		await this.page.locator(this.source).fill('Insured');
		//await this.page.waitForTimeout(1000);
		await this.page.locator(this.reason).click();
		await this.page.locator(this.reason).press('Delete');
		//await this.page.waitForTimeout(1000);
		await this.page.locator(this.reason).click();
		await this.page.locator(this.reason).fill('Policy not-taken');
		await this.page.locator(this.reason).press('Enter');
		await this.page.locator(this.refund_Method).click();
		await this.page.locator(this.start_CancellationBtn).click();
		//await this.page.waitForTimeout(3000);
		await this.page.locator(this.bind_Option).click();
		await this.page.locator(this.cancel_Now).click();
		await this.page.locator(this.warning_Message_Ok).click();
		const tittleStatus = await this.page.locator(this.tittle_Bar_Status).textContent();
		console.log("Tranaction Status is "+ tittleStatus);
		await expect (this.page.locator(this.tittle_Bar_Status)).toContainText('Cancellation Bound');
		//await this.page.waitForTimeout(1000);
		await this.page.locator(this.view_Your_Policy).click();
		await this.page.screenshot({ path: 'Policy Cancel.png' });
		await this.page.locator(this.policy_Transactions).click();
		await expect (this.page.locator(this.policy_Current_Status)).toContainText('Canceled'); 

	}

}

File: src/main/java/org/POM/PolicyCancel.java
package org.POM;

public class PolicyCancel {
	
	private String actions="//*[@id='PolicyFile:PolicyFileMenuActions-btnInnerEl']";
	private String cancel_Policy="//*[@id='PolicyFile:PolicyFileMenuActions:PolicyFileMenuActions_NewWorkOrder:PolicyFileMenuActions_CancelPolicy-textEl']";
	private String source="//*[@id='StartCancellation:StartCancellationScreen:CancelPolicyDV:Source-inputEl']";
	private String reason="//*[@id='StartCancellation:StartCancellationScreen:CancelPolicyDV:Reason-inputEl']";
	private String reason_Description="//*[@id='StartCancellation:StartCancellationScreen:CancelPolicyDV:ReasonDescription-inputEl']";
	private String refund_Method="//*[@id='StartCancellation:StartCancellationScreen:CancelPolicyDV:CalcMethod-inputEl']";
	private String start_CancellationBtn="//*[@id='StartCancellation:StartCancellationScreen:NewCancellation-btnInnerEl']";
	private String bind_Option="//*[@id='CancellationWizard:CancellationWizard_QuoteScreen:JobWizardToolbarButtonSet:BindOptions-btnInnerEl']";
	private String cancel_Now="//*[@id='CancellationWizard:CancellationWizard_QuoteScreen:JobWizardToolbarButtonSet:BindOptions:CancelNow-textEl']";
	private String warning_Message_Ok="//*[text()='OK']";
	private String tittle_Bar_Status="//*[@id='JobComplete:JobCompleteScreen:ttlBar']";
	private String view_Your_Policy="//*[@id='JobComplete:JobCompleteScreen:JobCompleteDV:ViewPolicy-inputEl']";
	private String policy_Transactions="//*[@id='PolicyFile:MenuLinks:PolicyFile_PolicyFile_Jobs']";
	private String policy_Current_Status="//div[@id='PolicyFile_Jobs/DetailsTitleCard']//table//table[1]//tr//td[5]//div";
	
	public String getActions() {
		return actions;
	}
	public void setActions(String actions) {
		this.actions = actions;
	}
	public String getCancel_Policy() {
		return cancel_Policy;
	}
	public void setCancel_Policy(String cancel_Policy) {
		this.cancel_Policy = cancel_Policy;
	}
	public String getSource() {
		return source;
	}
	public void setSource(String source) {
		this.source = source;
	}
	public String getReason() {
		return reason;
	}
	public void setReason(String reason) {
		this.reason = reason;
	}
	public String getReason_Description() {
		return reason_Description;
	}
	public void setReason_Description(String reason_Description) {
		this.reason_Description = reason_Description;
	}
	public String getRefund_Method() {
		return refund_Method;
	}
	public void setRefund_Method(String refund_Method) {
		this.refund_Method = refund_Method;
	}
	public String getStart_CancellationBtn() {
		return start_CancellationBtn;
	}
	public void setStart_CancellationBtn(String start_CancellationBtn) {
		this.start_CancellationBtn = start_CancellationBtn;
	}
	public String getBind_Option() {
		return bind_Option;
	}
	public void setBind_Option(String bind_Option) {
		this.bind_Option = bind_Option;
	}
	public String getCancel_Now() {
		return cancel_Now;
	}
	public void setCancel_Now(String cancel_Now) {
		this.cancel_Now = cancel_Now;
	}
	public String getWarning_Message_Ok() {
		return warning_Message_Ok;
	}
	public void setWarning_Message_Ok(String warning_Message_Ok) {
		this.warning_Message_Ok = warning_Message_Ok;
	}
	public String getTittle_Bar_Status() {
		return tittle_Bar_Status;
	}
	public void setTittle_Bar_Status(String tittle_Bar_Status) {
		this.tittle_Bar_Status = tittle_Bar_Status;
	}
	public String getView_Your_Policy() {
		return view_Your_Policy;
	}
	public void setView_Your_Policy(String view_Your_Policy) {
		this.view_Your_Policy = view_Your_Policy;
	}
	public String getPolicy_Transactions() {
		return policy_Transactions;
	}
	public void setPolicy_Transactions(String policy_Transactions) {
		this.policy_Transactions = policy_Transactions;
	}
	public String getPolicy_Current_Status() {
		return policy_Current_Status;
	}
	public void setPolicy_Current_Status(String policy_Current_Status) {
		this.policy_Current_Status = policy_Current_Status;
	}
	
	

}
